lib/natalie/compiler/pass0c.rb

```diff
@@ -53,6 +132,22 @@ def canary_array_max(exp, values)
         return exp.new(:if, exp.new(:call, first, :>, second), canary_array_max(exp, [first, *rest]), canary_array_max(exp, [second, *rest]))
       end
 
+      def canary_const_op(exp, values)
+        _, (_, arg1), op, (_, arg2) = exp
+        return s(:lit, arg1.send(op, arg2))
+      end
+      def canary_const_bool_op(exp,values)
+        _, (_, arg1), op, (_, arg2) = exp
+        return s(:true) if arg1.send(op, arg2)
+        return s(:false)
+      end
+
+      def canary_const_not(exp, values)
+        _, (type,*), _not = exp
+        return s(:false) if type_is_always_truthy(type)
```

Comment:
When using 's(...)' you may want to use instead 'exp.new(...)' as that will propagate the original source metadata (file, line)

Reply:
ahh ok didn't know that
will change that accordingly